tfhe/src/integer/server_key/radix_parallel/tests_unsigned/test_aes.rs

```diff
@@ -0,0 +1,225 @@
+            ^ gmul(original[col + 3], 2);
+    }
+}
+fn add_round_key(state: &mut [u8; 16], round_key: u128) {
+    let key_bytes = round_key.to_be_bytes();
+    for i in 0..16 {
+        state[i] ^= key_bytes[i];
+    }
+}
+fn plain_aes_encrypt_block(block_bytes: &mut [u8; 16], expanded_keys: &[u128]) {
+    add_round_key(block_bytes, expanded_keys[0]);
+    for round_key in expanded_keys.iter().take(10).skip(1) {
+        sub_bytes(block_bytes);
+        shift_rows(block_bytes);
+        mix_columns(block_bytes);
+        add_round_key(block_bytes, *round_key);
+    }
+    sub_bytes(block_bytes);
+    shift_rows(block_bytes);
+    add_round_key(block_bytes, expanded_keys[10]);
+}
+fn plain_aes_ctr(num_aes_inputs: usize, iv: u128, key: u128) -> Vec<u128> {
+    let expanded_keys = plain_key_expansion(key);
+    let mut results = Vec::with_capacity(num_aes_inputs);
+    for i in 0..num_aes_inputs {
+        let counter_value = iv.wrapping_add(i as u128);
+        let mut block = counter_value.to_be_bytes();
+        plain_aes_encrypt_block(&mut block, &expanded_keys);
+        results.push(u128::from_be_bytes(block));
+    }
+    results
+}
+
+fn internal_aes_fixed_parallelism_test<P, E>(param: P, mut executor: E, num_aes_inputs: usize)
+where
+    P: Into<TestParameters>,
+    E: for<'a> FunctionExecutor<
+        (&'a RadixCiphertext, &'a RadixCiphertext, u128, usize, usize),
+        RadixCiphertext,
+    >,
+{
+    let param = param.into();
+    let (cks, sks) = KEY_CACHE.get_from_params(param, IntegerKeyKind::Radix);
+    let cks = RadixClientKey::from((cks, 1));
+    let sks = Arc::new(sks);
+    executor.setup(&cks, sks);
+
+    let key: u128 = 0x2b7e151628aed2a6abf7158809cf4f3c;
+    let iv: u128 = 0xf0f1f2f3f4f5f6f7f8f9fafbfcfdfeff;
+
+    let plain_results = plain_aes_ctr(num_aes_inputs, iv, key);
+
+    let ctxt_key = cks.encrypt_u128_for_aes_ctr(key);
+    let ctxt_iv = cks.encrypt_u128_for_aes_ctr(iv);
+
+    for sbox_parallelism in [1, 2, 4, 8, 16] {
+        let encrypted_result =
+            executor.execute((&ctxt_key, &ctxt_iv, 0, num_aes_inputs, sbox_parallelism));
+        let fhe_results = cks.decrypt_u128_from_aes_ctr(&encrypted_result, num_aes_inputs);
+        assert_eq!(fhe_results, plain_results);
+    }
+}
+
+pub fn aes_fixed_parallelism_1_input_test<P, E>(param: P, executor: E)
+where
+    P: Into<TestParameters>,
+    E: for<'a> FunctionExecutor<
+        (&'a RadixCiphertext, &'a RadixCiphertext, u128, usize, usize),
+        RadixCiphertext,
+    >,
+{
+    internal_aes_fixed_parallelism_test(param, executor, 1);
+}
+
+pub fn aes_fixed_parallelism_2_inputs_test<P, E>(param: P, executor: E)
+where
+    P: Into<TestParameters>,
+    E: for<'a> FunctionExecutor<
+        (&'a RadixCiphertext, &'a RadixCiphertext, u128, usize, usize),
+        RadixCiphertext,
+    >,
+{
+    internal_aes_fixed_parallelism_test(param, executor, 2);
+}
+
+pub fn aes_dynamic_parallelism_many_inputs_test<P, E>(param: P, mut executor: E)
```

Comment:
Add some comments explaining what this test is testing. i.e., what is "dynamic parallelism".